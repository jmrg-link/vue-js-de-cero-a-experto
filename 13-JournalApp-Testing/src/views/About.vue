<template>
  <div class="about">
    <h1>This is an about page</h1>
    <h2>Actualizar snapshot</h2>
  </div>
</template>
