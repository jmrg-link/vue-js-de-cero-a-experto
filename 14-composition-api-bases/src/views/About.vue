<template>
	<div class="about">
		<h1>{{ title }}</h1>
		<button @click="changeTitle">Cambiar titulo</button>
	</div>
</template>

<script>
import {
	onBeforeMount,
	onMounted,
	onUpdated,
	onBeforeUnmount,
	onUnmounted,
	onErrorCaptured,
	onRenderTracked,
	onRenderTriggered,
	onActivated,
	onDeactivated,
	ref
} from 'vue'

export default {
	setup () {
		const title = ref ( 'about' )
		
		onMounted ( () => {
			console.log ( 'onMounted' )
		} )
		onBeforeMount ( () => {
			console.log ( 'onBeforeMount' )
		} )
		onUpdated ( () => {
			console.log ( 'onUpdated' )
		} )
		onBeforeUnmount ( () => {
			console.log ( 'onBeforeUnmount' )
		} )
		onUnmounted ( () => {
			console.log ( 'onUnmounted' )
		} )
		onErrorCaptured ( () => {
			console.log ( 'onErrorCaptured' )
		} )
		onRenderTracked ( () => {
			console.log ( 'onRenderTracked' )
		} )
		onRenderTriggered ( () => {
			console.log ( 'onRenderTriggered' )
		} )
		onActivated ( () => {
			console.log ( 'onActivated' )
		} )
		onDeactivated ( () => {
			console.log ( 'onDeactivated' )
		} )
		
		return {
			title,
			changeTitle: () => title.value = 'New About'
		}
	},
}
</script>