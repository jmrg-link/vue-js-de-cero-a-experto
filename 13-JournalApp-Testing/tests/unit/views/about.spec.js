import { shallowMount } from "@vue/test-utils";
import About from "@/views/About.vue"

describe("Pruebas en el About.vue", () => {
    test('Debe de renderizar el componente correctamente', () => {
        const wrapper = shallowMount(About)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
