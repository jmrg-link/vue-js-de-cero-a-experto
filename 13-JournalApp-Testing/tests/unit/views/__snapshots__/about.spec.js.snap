// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el About.vue Debe de renderizar el componente correctamente 1`] = `
<div class="about">
  <h1>This is an about page</h1>
  <h2>Actualizar snapshot</h2>
</div>
`;
